### ISSUE TITLE
Confusing `prelude-format-on-save` setting

### ISSUE THREAD

**Issue by knilink** [bbatsov/prelude#1421]
Hi, having been using this config for nearly a decade, thank you for creating and maintaining this project and i very appreciate it.

It's not a bug but more like a usability issue which my prettier-mod's format-on-save kept getting overrided by tide and `(remove-hook 'before-save-hook 'tide-format-before-save)` didn't stop it from happening. I spent quite sometime investigating an wrong direction until I global search the entire config and discover that it's because the function is wrapped with lambda.

https://github.com/bbatsov/prelude/blob/9106aea3413c92bc8c4e9c3421efc980e1dcb6e3/modules/prelude-ts.el#L46-L50

`prelude-format-on-save` seems to only work with prelude-ts and I think this feature is more like a "personal" config, and if we do want to keep this feature, a **toggle function** to add/remove the hook, instead of a boolean value, would probably be a better approach.

related issue https://github.com/bbatsov/prelude/pull/1254

## Expected behavior
`(remove-hook 'before-save-hook 'tide-format-before-save t)` should disable `tide-format-before-save`

## Actual behavior
It doesn't

**Comment by bbatsov** [comment]
I agree. The solution here seems pretty simple - we have make the hook function named, so it'd be easy to remove.
